src/shm_manager.cc

```diff
@@ -43,10 +43,13 @@ SharedMemoryManager::SharedMemoryManager(
 
   try {
     if (create) {
+      // Remove (if any) and create the region.
+      bi::shared_memory_object::remove(shm_region_name.c_str());
```

Comment:
@Tabrizian do we have any behavior/tests that expect a duplicate create() on an existing region to fail? Not sure if removing before every create would introduce a new/unexpected issue.

Reply:
We don't expect that behavior. I think its fine to remove the region before creating. But I'm not sure why @kthui has added this line. Do you see any failures if you remove this line?

Reply:
It is actually from a boost example: https://www.boost.org/doc/libs/1_49_0/doc/html/interprocess/sharedmemorybetweenprocesses.html#interprocess.sharedmemorybetweenprocesses.sharedmemory.shared_memory_a_simple_example

They remove the region when creating the `struct shm_remove` and remove again when it goes out of scope.

Reply:
I think they are just trying to be safe but either way its ok with me.